packages/ui-react-utils/src/renderLucideIconWithProps.ts

```diff
@@ -0,0 +1,111 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2015 - present Instructure, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+import React from 'react'
+import type { Renderable } from '@instructure/shared-types'
+import { callRenderProp } from './callRenderProp'
+
+/**
+ * Check if an element/component is a Lucide icon by checking its displayName
+ * @param element - Element or component to check
+ * @returns True if it's a Lucide icon (displayName starts with 'wrapLucideIcon(')
+ */
+const isLucideIcon = (element: any): boolean => {
+  if (React.isValidElement(element)) {
+    const displayName = (element.type as any)?.displayName
+    return displayName?.startsWith('wrapLucideIcon(')
+  }
+  return element?.displayName?.startsWith('wrapLucideIcon(')
```

Comment:
I don't understand this. If not valid element, why return this line? Why not return false? Can a valid lucide icon be a not valid element?